### ISSUE TITLE
NShader not working initially in VS2017

### ISSUE THREAD

**Issue by samizzo** [samizzo/nshader#13]
Split off from issue #12:

"I just installed NShader in 2017 and had the same problem as Xirema. I got it to work by disabling NShader, restarting Visual Studio, re-enabling Visual Studio and then restarting again."

**Comment by GabrielMajeri** [comment]
Hello, I am also suffering this issue and am unable to get NShader to work on VS2017. It worked well for a while, then it suddenly stopped working. I tried the workaround Xirema suggested and it still doesn't work.

I've followed [these instructions](https://blogs.msdn.microsoft.com/visualstudio/2010/02/24/troubleshooting-extensions-with-the-activity-log/) to get the activity log, and I found this relevant part of it:

```xml
<entry>
    <record>402</record>
    <time>2017/04/02 04:14:53.577</time>
    <type>Warning</type>
    <source>VisualStudio</source>
    <description>Unexpected system error mode before loading package [NShader]</description>
    <guid>{DD9B8CDE-96A0-4442-BF6F-083A7F43D453}</guid>
  </entry>
  <entry>
    <record>403</record>
    <time>2017/04/02 04:14:53.577</time>
    <type>Information</type>
    <source>VisualStudio</source>
    <description>Begin package load [NShader]</description>
    <guid>{DD9B8CDE-96A0-4442-BF6F-083A7F43D453}</guid>
  </entry>
  <entry>
    <record>404</record>
    <time>2017/04/02 04:14:53.593</time>
    <type>Information</type>
    <source>VisualStudio</source>
    <description>End package load [NShader]</description>
    <guid>{DD9B8CDE-96A0-4442-BF6F-083A7F43D453}</guid>
  </entry>
```

(this is the only part of the ActivityLog.xml file that contains any reference to NShader)

I am using the latest version of Visual Studio 2017 (26228.10) on Windows 10 x64. I tried reinstalling a few times and it still doesn't work.

**Comment by samizzo** [comment]
Thanks for the log! So you installed it, it was working, and then suddenly stopped? Did you have any older versions of NShader installed?

I'll try to find some time to look into it this week!

**Comment by GabrielMajeri** [comment]
Yes, I installed it, used it for a run or two, and then I opened a GLSL shader and noticed it wasn't working.

I didn't have any previous versions installed.

**Comment by GabrielMajeri** [comment]
I've installed the new release and it's fixed. Thanks!

**Comment by samizzo** [comment]
Oh thanks! I hadn't yet looked at this bug and wasn't sure if this would be fixed with the release. I'll do some more testing before I close this though.

**Comment by ShenyaoKe** [comment]
It works for me too after installing the new release of VS2017

**Comment by samizzo** [comment]
Great thanks!

**Comment by naqaden** [comment]
After installing only the newest version of NShader (v2.4), I also had this issue. I tested NShader while my previous shader extension (HLSL Tools) was disabled. To get NShader working for me, I had to fully uninstall HLSL.

**Comment by samizzo** [comment]
Thanks for the bug report!

Were you previously running both NShader and HLSL Tools together successfully?

**Comment by naqaden** [comment]
No problem. I started with only HLSL Tools, then disabled it before installing NShader. I never tried both enabled together.

**Comment by Rendman** [comment]
Same issue, installed fresh, syntax highlighting does not work.

**Comment by samizzo** [comment]
Did you get any error messages? Does it show up in Help->About?

What was the extension of the file you opened? How did you open it - via File->Open, or solution explorer, or drag/drop into VS, or some other means?

**Comment by Rendman** [comment]
No error messages. It does show up as ver 2.4 in Help-About. Extension is .vert (shader.vert), opened via solution explorer and was added by first creating a text file, then naming it shader.vert.

**Comment by krupitskas** [comment]
+1 here, vs2017. .vert and .frag not highlightning

**Comment by PazerOP** [comment]
+1

Hopefully there's no personal data in here. 
[ActivityLog.zip](https://github.com/samizzo/nshader/files/999661/ActivityLog.zip)

**Comment by samizzo** [comment]
Thanks for the info. In the mean time why don't you try [Hlsl Tools](https://github.com/tgjones/HlslTools). It has a lot more features than NShader. Honestly I'm not sure I'm even interested in continuing to support NShader now that Hlsl Tools exists.

**Comment by PazerOP** [comment]
Unfortunately I'm working with GLSL, so I chose to edit my shader files in VS Code with this extension: https://marketplace.visualstudio.com/items?itemName=slevesque.shader

**Comment by krupitskas** [comment]
Also GLSL user here
I'm working with Vulkan, so all my code in glsl

**Comment by samizzo** [comment]
You should be able to use Hlsl Tools with GLSL files. You can add extra file extensions for it to parse via Tools->Options->Text Editor->File Extension. Although it will complain about various GLSL types that it doesn't know about.

**Comment by Swyter** [comment]
I got it to work after disabling the [SPIRV-VSExtension](https://marketplace.visualstudio.com/items?itemName=SaschaWillems.SPIRV-VSExtension) by Sascha Willems. Seems like they were conflicting.

**Comment by ssteinberg** [comment]
Had the same problem. 
I decided to just take Samizzo's advice and adapt HLSL Tools to GLSL. I made a fork and quickly replaced HLSL's types and functions with GLSL ones (all 4.5 functions with parameters and documentations). It is far from a decent job, but it works. Will be maintained depending on available time.
https://github.com/ssteinberg/HlslTools

Anyway, thanks to Samizzo for keeping NShader alive, and Tim Jones for HLSL Tools.

**Comment by samizzo** [comment]
Thanks for letting us know about that! The NShader code base is quite old and dates back to VS 2008, and the Visual Studio extension SDK has changed over the years, so I guess that's why we've been having problems. I'll continue to tinker when I have time (working hard to release our game at the moment though!).

**Comment by realvictorprm** [comment]
@ssteinberg what about creating a seperate repo so we can maintain this :smiley:?

**Comment by ssteinberg** [comment]
Unlinked from HlslTools and renamed.
If anyone has the time to maintain, please let me know.

https://github.com/ssteinberg/GlslTools

**Comment by realvictorprm** [comment]
Thanks @ssteinberg. Yeah definitely needs maintenance

**Comment by realvictorprm** [comment]
Can you enable that issues can be opened?

**Comment by ssteinberg** [comment]
Done

**Comment by samizzo** [comment]
That's great, thanks for doing that!

**Comment by Alan-FGR** [comment]
Hey guys, I had this problem with the latest release, vanilla install. No biggie though.

**Comment by Markyparky56** [comment]
Would like to reiterate what @Swyter said, SPIRV-VSExtension seems to be conflicting/blocking NShader from kicking in.

**Comment by Alan-FGR** [comment]
@Markyparky56 I don't have that extension and still had the problem described here though.

**Comment by noemu** [comment]
I've got a conflict with the "Visual Studio 2017 Tools for Unity" Version: 3.1.0.0.

Deactivated it and NShader work again. (Can't say if Unity Extension is a part of vanilla VS)
